docker/pravega/scripts/init_kubernetes.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+#
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+
+k8() {
+    local namespace=$1
+    local resource_type=$2
+    local resource_name=$3
+    local jsonpath=$4
+    local bearer=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+    local cacert="/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+    if [ -z "$namespace" ]; then
+        retval=$( curl --cacert ${cacert} -H "Authorization: Bearer ${bearer}" https://kubernetes/api/v1/${resource_type}/${resource_name} 2> /dev/null | jq -rM "${jsonpath}" 2> /dev/null )
+    else
+        retval=$( curl --cacert ${cacert} -H "Authorization: Bearer ${bearer}" https://kubernetes/api/v1/namespaces/${namespace}/${resource_type}/${resource_name} 2> /dev/null | jq -rM "${jsonpath}" 2> /dev/null )
+    fi
+
+    echo "$retval"
+}
+
+init_kubernetes() {
+    if [ -z "${POD_NAMESPACE}" ]; then
```

Comment:
Given that `init_kubernetes` is invoked by `entrypoint.sh ` which is internally invoked at the start of the docker container( since it is the default `ENTRYPOINT` in the dockerfile). 
How is it ensured that `init_kubernetes` is invoked only in a K8 environment? 
(A normal user of the pravega container should not encounted POD_NAMESPACE variable not set)

Reply:
Great catch, I've provided a fix for this in #3129.